src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/AbstractRequest.java

```diff
@@ -438,13 +438,107 @@ public ResourceLocation getDisplayIcon()
     }
 
     @NotNull
+    @Override
+    public List<ItemStack> getDisplayStacks()
+    {
+        if (!(getRequest() instanceof IDeliverable))
+        {
+            return Lists.newArrayList();
+        }
+
+        final IDeliverable deliverable = (IDeliverable) getRequest();
+
+        if (itemExamples == null)
+        {
+            itemExamples = ImmutableList.copyOf(StreamSupport.stream(Spliterators.spliteratorUnknownSize(Item.REGISTRY.iterator(), Spliterator.ORDERED), false)
+                    .flatMap(item -> {
+                        NonNullList<ItemStack> stacks = NonNullList.create();
+                        try
+                        {
+                            item.getSubItems(item, null, stacks);
+                        }
+                        catch (Exception ex)
+                        {
+                            Log.getLogger().warn("Failed to get sub items from: " + item.getRegistryName());
+                        }
+
+                        return stacks.stream().filter(deliverable::matches);
+                    })
+                    .collect(Collectors.toList()));
+        }
+
+        return itemExamples;
+    }
+
+    @NotNull
+    @Override
     public <T> Optional<T> getRequestOfType(final Class<T> tClass)
     {
         final R request = getRequest();
         if (tClass.isInstance(request))
         {
             return Optional.of(tClass.cast(request));
         }
+
         return Optional.empty();
     }
+
+    @Override
+    public boolean equals(final Object o)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 10, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)